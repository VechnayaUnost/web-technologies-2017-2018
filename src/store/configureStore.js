import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

export default (initialState) => {
    const sagaMiddleware = createSagaMiddleware();
    const store = createStore(rootReducer, initialState, applyMiddleware(sagaMiddleware));

    sagaMiddleware.run(rootSaga);
    store.runSaga = () => sagaMiddleware.run(rootSaga);
    store.close = () => store.dispatch(END);

    return store;
};