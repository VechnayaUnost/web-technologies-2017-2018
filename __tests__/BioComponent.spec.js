import React from 'react';
import BioComponent from '../src/components/BioComponent';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('BioComponent component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        userInfo: {
            bio: "Neytiri te Tskaha Mo'at’ite is the Na'vi princess of the Omaticaya clan." +
                "She is the second-born daughter of Eytukan and Mo'at and younger sister of Sylwanin. " +
                "She meets Jake Sully in a Pandoran forest, helping him when he is attacked by a pack " +
                "of viperwolves. Neytiri soon becomes Jake's teacher and helps him to complete several " +
                "tasks. Eventually, they fall in love and mate under the Trees of Voices. Neytiri fights " +
                "alongside Jake in the Assault on the Tree of Souls and saves him from being killed by Miles " +
                "Quaritch. She will be the future Tsahìk of the clan, with her mate, Jake, as Olo'eyktan."
        }
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('BioComponent component rendered succesfully', () => {
        const bioComponent = mount(
            <BioComponent store={store} />
        );
        expect(mountToJson(bioComponent)).toMatchSnapshot();
    });

    it('BioComponent component props match state', () => {
        const bioComponent = shallow(
            <BioComponent store={store} />
        );
        expect(bioComponent.prop('bio')).toBe(initialState.userInfo.bio);
    })
});