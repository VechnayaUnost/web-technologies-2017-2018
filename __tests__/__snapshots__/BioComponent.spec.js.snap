// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BioComponent component test BioComponent component rendered succesfully 1`] = `
<Connect(BioComponent)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BioComponent
    bio="Neytiri te Tskaha Mo'at’ite is the Na'vi princess of the Omaticaya clan.She is the second-born daughter of Eytukan and Mo'at and younger sister of Sylwanin. She meets Jake Sully in a Pandoran forest, helping him when he is attacked by a pack of viperwolves. Neytiri soon becomes Jake's teacher and helps him to complete several tasks. Eventually, they fall in love and mate under the Trees of Voices. Neytiri fights alongside Jake in the Assault on the Tree of Souls and saves him from being killed by Miles Quaritch. She will be the future Tsahìk of the clan, with her mate, Jake, as Olo'eyktan."
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="BioComponent"
    >
      <h2>
        Neytiri te Tskaha Mo'at’ite is the Na'vi princess of the Omaticaya clan.She is the second-born daughter of Eytukan and Mo'at and younger sister of Sylwanin. She meets Jake Sully in a Pandoran forest, helping him when he is attacked by a pack of viperwolves. Neytiri soon becomes Jake's teacher and helps him to complete several tasks. Eventually, they fall in love and mate under the Trees of Voices. Neytiri fights alongside Jake in the Assault on the Tree of Souls and saves him from being killed by Miles Quaritch. She will be the future Tsahìk of the clan, with her mate, Jake, as Olo'eyktan.
      </h2>
    </div>
  </BioComponent>
</Connect(BioComponent)>
`;
